Change ontology term identifier

david4096 opened the pull request:
Ontology terms had an ambigious 'id' field and description. This PR renames the field to term_id to remove the confusion over the role of the field, as well as improves the comment for its proper use. A field 'term_uri' has been added that allows one to specify the full URI for a term if it is available.

Closes #621 #165

mbaudis commented:
+1 Clear improvement.

sarahhunt commented:
+1

mbaudis commented:
@david4096 (pls. don't forget the docs)

david4096 commented:
Thanks @mbaudis updated with some minor tweaks to the ontology documentation.

david4096 commented:
Related to discussion in #165

mcourtot commented:
+1 with small modification (with input from @cmungall): "if no CURIE are available." -> "if no URI available".

david4096 commented:
Thanks @mcourtot! Updated!

mbaudis commented:
Please check ```termId``` => ```term_id``` (if I understand the proto snake_case correctly); same for ```termUri``` => ```term_uri```. I usually do things like that with serarch/replace over the whole tree; but don't tell anybody, please.

If fixed -> merge.

david4096 commented:
Thanks @mbaudis good to go!

jmcmurry commented on doc/source/appendix/ontologies.rst:
```diff
@@ -38,30 +39,34 @@ What is the minimum attribute requirement for  OntologyTerm in GA4GH?
-  we assume this resolves to a meaningful document, e.g. http://purl.obolibrary.org/obo/SO_0000147
+:termId:
+  The unique identifier of the term in an ontology source. This should be a CURIE (e.g., SO:0001234), but may be an alphanumerical string (e.g., 8000/3) if no CURIE are available.
 :term:
```
In the interest of reducing ambiguity, it may be worthwhile to have this field be termLabel:

david4096 replied:
Perhaps we can go forward with `term_id` and address the ambiguity in a later PR. We have improved over the field named `id`, however, there may be further room for improvement.

jmcmurry commented on doc/source/appendix/ontologies.rst:
```diff
@@ -38,30 +39,34 @@ What is the minimum attribute requirement for  OntologyTerm in GA4GH?
 :term:
   preferred but not required (e.g. ‘exon’); corresponds to class label
+:termUri:
+  A fully qualified URI that can be used to access details regarding the term. e.g. http://purl.bioontology.org/ontology/SNOMEDCT/76345009.
```
If multiple discrepant URIs exist and are known, what is the plan for describing / collapsing them?

mcourtot replied:
I think we need to think about having the URI further - we have been discussing having a prefix registry (see https://github.com/ga4gh/schemas/issues/695) so this may be up for change/removal.

jmcmurry commented on doc/source/appendix/ontologies.rst:
```diff
@@ -38,30 +39,34 @@ What is the minimum attribute requirement for  OntologyTerm in GA4GH?
   not required since should be resolved from prefix etc., but supporting/fall-back in case of non-standard/deprecated/entropic annotations; possible use for CURIEs in compact sequence ontology implementations  (e.g. SO:0000147)
 :sourceVersion:
-  not required but good practice; if no explicit versioning, ISO8601 formatted data of retrieval should be used
+  not required but good practice; if no explicit versioning, ISO8601 formatted data of retrieval should be used.
```
Should there be an indication as to whether the date (if it exists) is the date of retrieval or the date of the release? Does it matter for your purposes? I don't have a particular horse in the race, just putting it out there for comment.

jmcmurry replied:
In some cases, there are versioned artifacts; it may be good to point out that these URIs too could be used in this field.eg.  http://purl.obolibrary.org/obo/uberon/releases/2016-01-26/uberon.owl

jmcmurry commented on doc/source/appendix/ontologies.rst:
```diff
@@ -84,122 +89,142 @@ Examples
+  "human_phenotype Ontology",
 :sourceVersion:
-	"release_Jan2016*"
+  "release_Jan2016*"
```
@drseb @cmungall what is the way that HP versions should be denoted?

drseb replied:
in the obo-file there is a "data-version" tag. I don't know right now how the annotation property is called in owl.

Sent from mobile

> On 08.09.2016, at 20:02, Julie McMurry <notifications@github.com> wrote:
> 
> In doc/source/appendix/ontologies.rst:
> 
> >  :sourceVersion:
> > -	"release_Jan2016*"
> > +  "release_Jan2016*"
> @drseb @cmungall what is the way that HP versions should be denoted?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

cmungall replied:
versionIRI

mcourtot replied:
+1 on using versionIRI for versioning (when available)

david4096 commented:
Thanks everyone for your input @jmcmurry @mcourtot @drseb ! I'll be happy to integrate any further work on this data model as consensus is reached.

For now, I'd like to see if we can move ahead and take further discussion to another issue. This PR was designed to improve the situation from `id` to `term_id` and improve the comments, which it has done. To continue the discussion check out the below links:

[Ontology Source Identification](https://github.com/ga4gh/schemas/issues/695)
[Multiple Ontology Term URIs](https://github.com/ga4gh/schemas/issues/708)
[OntologyTerm.id definition](https://github.com/ga4gh/schemas/issues/621)

jmcmurry commented:
That sounds good to me; thanks @david4096

cmungall commented:
And -10 to any date scheme that is not ISO-8601!

mbaudis commented:
@cmungall  @drseb ... but reminding me of https://github.com/phenopackets/phenopacket-format/issues/75

mbaudis commented:
@mcourtot @cmungall @david4096 This should now be updated to reflect the Vancouver discussions.

mcourtot commented:
Hm - I tried and push a commit to the pr/694 branch to reflect the Vancouver changes but don't see them here. Hadn't tried and commit over a PR before, so maybe i'm looking at the wrong place?

david4096 commented:
@mbaudis @mcourtot I've incorporated your changes from https://github.com/ga4gh/schemas/pull/726 .

Since we have removed the URLs, I believe we need a way to manage them, or at least specify a place where they might be looked up. What do you think @cmungall ?

mcourtot commented:
Thanks @david4096! We're planning on using OLS, http://www.ebi.ac.uk/ols/index to bind prefixes to their URIs. @simonjupp, @tburdett, do you have a specific link?

david4096 commented:
Implemented in the server [here](https://github.com/ga4gh/server/pull/1523).
